shell/plugins/back-button.js

```diff
@@ -1,3 +1,12 @@
-window.addEventListener('popstate', () => {
-  window._popStateDetected = true;
-});
+export default function({ app }) {
+  window.addEventListener('popstate', () => {
+    window._popStateDetected = true;
+  });
+
+  app.router.afterEach((to, from) => {
```

Comment:
Could we have router-related logic inside the router configuration?
Sparse code is quite hard to track.